File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/FeedbackAnalysisResults/Models/FeedbackAnalysisResultFilter.cs
using Feedback.Analyzer.Domain.Common.Query;

namespace Feedback.Analyzer.Application.FeedbackAnalysisResults.Models;

/// <summary>
/// Represents a filter for querying feedback analysis results with pagination support.
/// </summary>
public class FeedbackAnalysisResultFilter : FilterPagination
{
    /// <summary>
    /// Initializes a new instance of the <see cref="FeedbackAnalysisResultFilter"/> class.
    /// </summary>
    public FeedbackAnalysisResultFilter()
    {
        PageSize = int.MaxValue;
        PageToken = 1;
    }
    
    /// <summary>
    /// Computes the hash code for the filter.
    /// </summary>
    /// <returns>The hash code value for the filter.</returns>
    public override int GetHashCode()
    {
        var hashCode = new HashCode();

        hashCode.Add(PageSize);
        hashCode.Add(PageToken);

        return hashCode.ToHashCode();
    }
    
    /// <summary>
    /// Determines whether the specified object is equal to the current filter.
    /// </summary>
    /// <param name="obj">The object to compare with the current filter.</param>
    /// <returns>True if the specified object is equal to the current filter; otherwise, false.</returns>
    public override bool Equals(object? obj)
    {
        return obj is FeedbackAnalysisResultFilter feedback && feedback.GetHashCode() == GetHashCode();
    }
}

File: src/Insight.Box.FrontEnd/src/infrastructure/apiClients/insightBoxClient/brokers/FeedbackAnalysisResultFilter.ts
import {FilterPagination} from "@/common/FilterPagination";

/*
 * Represents workflow results filter
 */
export class FeedbackAnalysisResultFilter extends  FilterPagination {

    /*
     * Workflow id
     */
    public workflowId: string;

    /*
     * Result id
     */
    public resultId: string;

    constructor(workflowId: string, resultId: string) {
        super();
        this.pageSize = 5;
        this.workflowId = workflowId;
        this.resultId = resultId;

        this.searchKeyword = '';
    }
}